hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/push/DelayedEventBusPushStrategy.java

```diff
@@ -163,38 +173,43 @@ private DispatchRunnable(final UI ui, final VaadinSession session) {
 
         @Override
         public void run() {
-            LOG.debug("UI EventBus aggregator started");
+            LOG.debug("UI EventBus aggregator started for UI {}", uiid);
             final long timestamp = System.currentTimeMillis();
-            final List<org.eclipse.hawkbit.eventbus.event.Event> events = new LinkedList<>();
-            for (int i = 0; i < BLOCK_SIZE; i++) {
+
+            final int size = queue.size();
+            if (size <= 0) {
+                LOG.debug("UI EventBus aggregator for UI {} has nothing to do.", uiid);
+                return;
+            }
+
+            final List<org.eclipse.hawkbit.eventbus.event.Event> events = new ArrayList<>(size);
+            for (int i = 0; i < size; i++) {
                 final org.eclipse.hawkbit.eventbus.event.Event pollEvent = queue.poll();
                 if (pollEvent == null) {
                     continue;
                 }
-                events.add(pollEvent);
+                events.add(i, pollEvent);
```

Comment:
If the order is correct better go for `queue.drainTo(...)` due to:
> This operation may be more efficient than repeatedly polling this queue

If you'll change the order as mentioned by @michahirsch take into account to use a `Deque` instead of a handcrafted reverse-loop approach.